src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -0,0 +1,602 @@
+import org.ow2.proactive.connector.iaas.model.ScriptResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Lists;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.OperatingSystemTypes;
+import com.microsoft.azure.management.compute.VirtualMachine;
+import com.microsoft.azure.management.compute.VirtualMachineCustomImage;
+import com.microsoft.azure.management.compute.VirtualMachineDataDisk;
+import com.microsoft.azure.management.compute.VirtualMachineExtension;
+import com.microsoft.azure.management.compute.VirtualMachineSizeTypes;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.network.NetworkInterface;
+import com.microsoft.azure.management.network.NetworkSecurityGroup;
+import com.microsoft.azure.management.network.PublicIpAddress;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.utils.SdkContext;
+
+import lombok.Getter;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 01/03/17
+ */
+@Component
+public class AzureProvider implements CloudProvider {
+
+    private final Logger logger = Logger.getLogger(AzureProvider.class);
+
+    @Getter
+    private final String type = "azure";
+
+    private static final Region defaultRegion = Region.US_EAST;
+
+    private static final VirtualMachineSizeTypes defaultVMSize = VirtualMachineSizeTypes.STANDARD_D1_V2;
+
+    private static final int resourcesNameExtraChars = 5;
+
+    private static final String virtualNetworkNameBase = "vnet";
+
+    private static final String publicIPAddressNameBase = "ip";
+
+    private static final String networkSecurityGroupNameBase = "sg";
+
+    private static final String networkInterfaceNameBase = "if";
+
+    private static final String osDiskNameBase = "os";
+
+    private static final String defaultUsername = "activeeon";
+
+    private static final String defaultPassword = "Act1v€0N";
+
+    private static final String defaultPrivateNetworkCIDR = "10.0.0.0/24";
+
+    private static final Boolean defaultStaticPublicIP = true;
+
+    private static final String scriptExtensionPublisher = "Microsoft.Azure.Extensions";
+
+    private static final String scriptExtensionType = "CustomScript";
+
+    private static final String scriptExtensionVersion = "2.0";
+
+    private static final String scriptExtensionCmdKey = "commandToExecute";
+
+    @Autowired
+    private AzureServiceCache azureServiceCache;
+
+    @Autowired
+    private AzureProviderUtils azureProviderUtils;
+
+    @Override
+    public Set<Instance> createInstance(Infrastructure infrastructure, Instance instance) {
+
+        Azure azureService = azureServiceCache.getService(infrastructure);
```

Comment:
Unfortunately, there is no mention of thread safety in the documentation, after some research I found a post from a Microsoft dev saying:
```
We do not make any thread-safety guarantees about our libraries. We also do not test them for thread-safety. Methods that are currently thread-safe may be thread-unsafe in future versions. Thank you for pointing this out to us. We will look into updating our documentation.
```
Anyway it seems that, in practice, users found that most of the classes are actually thread-safe.
After a few tests on my side, I can confirm that createInstance and deleteInstance are indeed tread-safe. But anyway I will mention the uncertainty about future upgrades into the class header.